### DISCUSSION TITLE
Apply rate limit to query and mutations

### DISCUSSION THREAD

**Discussion by samarthhegdekalgar** [TanStack/query#2779] [Q&A]
Hello, we are facing a unique problem with API's, our backend has rate limit for all the mutations that means we can call any particular mutation only once per second, and we have a multiple mutations which will be called many times within a second. I wonder react-query has any options where we can control the api call and avoid API calls to overcome from rate limit problem.

**Comment by rusakovic** [comment]
support it. Will be very helpful!

**Comment by lkostrowski** [comment]
+1

It's quite popular that many 3rd party APIs implement rate limits, would be great to handle it at client level

**Comment by bzbetty** [comment]
Had a useEffect vs useMutation bug that spammed my server hard (couple of thousand requests in a few seconds).  Some kind of client side rate limit would be nice to help prevent this.
